contracts/CurrencyNetworkRegistry.sol

```diff
@@ -0,0 +1,95 @@
+pragma solidity ^0.5.8;
+
+import "./CurrencyNetworkInterface.sol";
+import "./lib/ERC165Query.sol";
+
+
+contract CurrencyNetworkRegistry is ERC165Query {
+
+    struct CurrencyNetworkMetadata {
+        address author;
+        string name;
+        string symbol;
+        uint8 decimals;
+    }
+
+    mapping (address => CurrencyNetworkMetadata) internal currencyNetworks;
+    address[] internal registeredCurrencyNetworks;
+
+    event CurrencyNetworkAdded(
+        address indexed _address,
+        address _author,
+        string _name,
+        string _symbol,
+        uint8 _decimals
+    );
+
+    function addCurrencyNetwork(address _address) external {
+        CurrencyNetworkInterface network;
+
+        require(
+            currencyNetworks[_address].author == address(0) &&
+            bytes(currencyNetworks[_address].name).length == 0 &&
+            bytes(currencyNetworks[_address].symbol).length == 0,
+            "CurrencyNetworks can only be registered once."
+        );
+
+        network = CurrencyNetworkInterface(_address);
```

Comment:
What happens if `_address` does not implement the interface?

Reply:
The doesContractImplementInterface check makes sure that the interface is fulfilled. I only create the instance before, because I need the function signatures to calculate the hash. No call is made to the contract at this point.
But yes, if something went wrong, it would probably just silently fail.